server/events.go

```diff
@@ -1184,7 +1184,11 @@ func (s *Server) sendAccConnsUpdate(a *Account, subj ...string) {
 			a.ctmr.Reset(eventsHBInterval)
 		}
 	}
+	for _, sub := range subj {
+		sendQ <- &pubMsg{nil, sub, _EMPTY_, &m.Server, &m, false}
```

Comment:
Yeah, but I could see how this whole thing could deadlock (may not be the only case, but...): suppose the send queue is full (because other go routines keep trying to add to the queue) and the internalSendLoop() is currently processing 1 message and as part of processInboundClientMsg() is locking this account. Then we have a problem.

Reply:
Agreed, was hoping the queue size would avoid that. 

The other alternative is to let go of the account lock earlier. 
This will improve the test as well (essentially moving the timer removal prior to send), but it will not rule out a test failure.  (just makes it more unlikely)

What do you think about unlocking if the send where to block?
```
select {
case sendQ <- &pubMsg{nil, sub, _EMPTY_, &m.Server, &m, false}:
default:
    a.mu.Unlock()
    sendQ <- &pubMsg{nil, sub, _EMPTY_, &m.Server, &m, false}:
    a.mu.Lock()
}
```

Reply:
That's indeed better than blocking with the account lock held and risking a deadlock when queue is full.

Reply:
Will do that then. This will avoid event reordering until the queue gets full (which really it shouldn't)